react/IOMessageWithMarkers.tsx

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable react/display-name */
+import React, { useMemo } from 'react'
+import { useCssHandles } from 'vtex.css-handles'
+
+import IOMessage from './IOMessage'
+import { IOMessageWithMarkers as IOMessageWithMarkersType } from './typings/IOMessageWithMarkers'
+
+const IOMessageWithMarkers: IOMessageWithMarkersType = ({
+  label,
+  markers = [],
+  componentName,
+  values = {},
+}) => {
+  const CSS_HANDLES = useMemo(() => markers.map(marker => {
+    return `${componentName}_${marker}`
+  }), [markers])
+  const handles = useCssHandles(CSS_HANDLES)
+  
+  const markerComponents = markers.reduce((acc: any, marker) => {
+    acc[marker] = (...chunks: any) => (
+      <span className={handles[`${componentName}_${marker}`]}>{chunks}</span>
+    )
+    return acc
+  }, {})
+
+  return (
+    <IOMessage
+      id={label}
+      values={{
+        ...markerComponents,
+        ...values,
+      }}
+    />
+  )
+}
+
+export default React.memo(IOMessageWithMarkers)
```

Comment:
```suggestion
const IOMessageWithMarkers: IOMessageWithMarkersType = React.memo(function IOMessageWithMarkers(props) {
  const {
    label,
    markers = [],
    componentName,
    values = {},
  } = props
  const CSS_HANDLES = useMemo(() => markers.map(marker => {
    return `${componentName}_${marker}`
  }), [markers])
  const handles = useCssHandles(CSS_HANDLES)
  
  const markerComponents = markers.reduce((acc: any, marker) => {
    acc[marker] = (...chunks: any) => (
      <span className={handles[`${componentName}_${marker}`]}>{chunks}</span>
    )
    return acc
  }, {})

  return (
    <IOMessage
      id={label}
      values={{
        ...markerComponents,
        ...values,
      }}
    />
  )
})

export default IOMessageWithMarkers
```

Reply:
I think iaron's version is a tiny bit more readable, but both are ok

Reply:
I prefer how it is right now too